packages/mui-styles/src/makeStyles/makeStyles.test.js

```diff
@@ -44,72 +23,73 @@ describe('makeStyles', () => {
-    const extendedClasses = output.classes;
+    const useStyles = makeStyles(styles);
+
+    const { result, rerender } = renderHook((props) => useStyles(props));
+    const baseClasses = result.current;
+    rerender({ classes: { root: undefined } });
+    const extendedClasses = result.current;
+
     expect(extendedClasses.root).to.equal(baseClasses.root);
   });
 
   describe('warnings', () => {
-    const mountWithProps = createGetClasses({ root: {} });
+    const useStyles = makeStyles({ root: {} });
 
     it('should warn if providing a unknown key', () => {
-      const output = mountWithProps();
+      let baseClasses;
+      let extendedClasses;
 
       expect(() => {
-        output.wrapper.setProps({ classes: { bar: 'foo' } });
+        const { result } = renderHook(() => useStyles({ classes: { bar: 'foo' } }));
+        baseClasses = result.current;
+        extendedClasses = result.current;
       }).toErrorDev('MUI: The key `bar` provided to the classes prop is not implemented');
-
-      const baseClasses = output.classes;
-      const extendedClasses = output.classes;
       expect(extendedClasses).to.deep.equal({ root: baseClasses.root, bar: 'undefined foo' });
     });
 
     it('should warn if providing a string', () => {
-      const output = mountWithProps();
-
       expect(() => {
-        output.wrapper.setProps({ classes: 'foo' });
+        renderHook(() => useStyles({ classes: 'foo' }));
       }).toErrorDev(['You might want to use the className prop instead']);
     });
 
     it('should warn if providing a non string', () => {
-      const output = mountWithProps();
-      const baseClasses = output.classes;
+      const { result, rerender } = renderHook((props) => useStyles(props));
+      const baseClasses = result.current;
 
       expect(() => {
-        output.wrapper.setProps({ classes: { root: {} } });
+        rerender({ classes: { root: {} } });
       }).toErrorDev('MUI: The key `root` provided to the classes prop is not valid');
 
-      const extendedClasses = output.classes;
+      const extendedClasses = result.current;
+
       expect(extendedClasses).to.deep.equal({ root: `${baseClasses.root} [object Object]` });
     });
 
     it('should warn if missing theme', () => {
-      const styles = (theme) => ({ root: { padding: theme.spacing(2) } });
-      const mountWithProps2 = createGetClasses(styles);
+      const useStyles2 = makeStyles((theme) => ({ root: { padding: theme.spacing(2) } }));
 
       expect(() => {
         expect(() => {
-          mountWithProps2({});
+          renderHook(() => useStyles2({}));
         }).to.throw('theme.spacing is not a function');
       }).toErrorDev([
+        'MUI: The `styles` argument provided is invalid.\nYou are providing a function without a theme in the context.',
+        'MUI: The `styles` argument provided is invalid.\nYou are providing a function without a theme in the context.',
+        'Uncaught [TypeError: theme.spacing is not a function',
```

Comment:
Not sure why errors are now duplicated, probably due to effects running twice now that `renderHook` is used. We [pass](https://github.com/mui/material-ui/pull/42703/files?diff=unified&w=1#diff-8b225c83d49a250a760df0f45d2baac84945f674f71d752dd958fa1d97374675R16) `strict: false` to `createRenderer` (and to `createMount` before). Maybe we could look into creating a our own `renderHook` wrapper with this option.

Reply:
This package is not compatible with strict mode, so yes, we need to have that option.